code/controllers/subsystem/minimaps.dm

```diff
@@ -0,0 +1,281 @@
+		charlieupdators.minimap.overlays += minimaps_by_z["[charlieupdators.ztarget]"].charlie_images_raw
+	for(var/datum/minimap_updator/deltaupdators AS in update_targets[MINIMAP_STRING_DELTA])
+		deltaupdators.minimap.overlays += minimaps_by_z["[deltaupdators.ztarget]"].delta_images_raw
+
+/**
+ * Adds an atom to the processing updators that will have blips drawn on them
+ * Arguments:
+ * * target: the target we want to be updating the overlays on
+ * * flags: flags for the types of blips we want to be updated
+ * * ztarget: zlevel we want to be updated with
+ */
+/datum/controller/subsystem/minimaps/proc/add_to_updaters(atom/target, flags, ztarget)
+	var/datum/minimap_updator/holder = new(target, ztarget)
+	if(flags & MINIMAP_FLAG_XENO)
+		LAZYADD(update_targets[MINIMAP_STRING_XENO], holder)
+	if(flags & MINIMAP_FLAG_MARINE)
+		LAZYADD(update_targets[MINIMAP_STRING_MARINE], holder)
+	if(flags & MINIMAP_FLAG_ALPHA)
+		LAZYADD(update_targets[MINIMAP_STRING_ALPHA], holder)
+	if(flags & MINIMAP_FLAG_BRAVO)
+		LAZYADD(update_targets[MINIMAP_STRING_BRAVO], holder)
+	if(flags & MINIMAP_FLAG_CHARLIE)
+		LAZYADD(update_targets[MINIMAP_STRING_CHARLIE], holder)
+	if(flags & MINIMAP_FLAG_DELTA)
+		LAZYADD(update_targets[MINIMAP_STRING_DELTA], holder)
+	updators_by_datum[target] = holder
+	update_targets_unsorted += target
+	RegisterSignal(target, COMSIG_PARENT_QDELETING, .proc/remove_updator)
+
+/**
+ * Removes a atom from the subsystems updating overlays
+ */
+/datum/controller/subsystem/minimaps/proc/remove_updator(atom/target)
+	SIGNAL_HANDLER
+	var/datum/minimap_updator/holder = updators_by_datum[target]
+	updators_by_datum -= target
+	LAZYREMOVE(update_targets[MINIMAP_STRING_XENO], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_MARINE], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_ALPHA], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_BRAVO], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_CHARLIE], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_DELTA], holder)
+	update_targets_unsorted -= target
+
+/**
+ * Holder datum for a zlevels data, concerning the overlays and the drawn level itself
+ * The individual image trackers have a raw and a normal list
+ * raw lists just store the images, while the normal ones are assoc list of [tracked_atom] = image
+ * the raw lists are to speed up the Fire() of the subsystem so we dont have to filter through
+ */
+/datum/hud_displays
+	///Actual icon of the drawn zlevel with all of it's atoms
+	var/icon/hud_image
+	var/list/xeno_images = list()
+	var/list/xeno_images_raw = list()
+	var/list/marine_images = list()
+	var/list/marine_images_raw = list()
+	var/list/alpha_images = list()
+	var/list/alpha_images_raw = list()
+	var/list/bravo_images = list()
+	var/list/bravo_images_raw = list()
+	var/list/charlie_images = list()
+	var/list/charlie_images_raw = list()
+	var/list/delta_images = list()
+	var/list/delta_images_raw = list()
```

Comment:
can't you make an assoc list for those ?

Reply:
Bitfields cant be used as keys: https://discord.com/channels/484170914754330625/487268744419344384/816992021326725140